### DISCUSSION TITLE
замечания и предложения по диалогу Ксардаса

### DISCUSSION THREAD

**Discussion by SpectresHub** [dosinabox/g2nr_unofficial_update#530] [General]
- [x] функции условия `DIA_Xardas_FirstEXIT_Condition()`, `DIA_Xardas_TODO_Condition()`, `DIA_Xardas_Khorinis_Condition()`:
 `(Kapitel < 3)` - не нужно

- [x] функция условия `DIA_Xardas_AWAY_Condition()`:
`!Npc_KnowsInfo(other,DIA_Xardas_FirstEXIT) && (Kapitel < 3)` - не нужно

- [ ] функция диалога `DIA_Addon_Xardas_Hello_Man()`:
я бы убрал добавленную строку `AI_Output(other,self,"DIA_Addon_AddonIntro_15_00");`
реплика ГГ выглядит излишней, **монолог** Ксардаса звучит более естественно

- [ ] функция диалога `DIA_Addon_Xardas_StonePlate_Info()`:
ГГ может иметь в инвентаре любую каменную табличку, ту что в башне Ксардаса может вообще не брать, в таком случае диалог будет звучать странно.
2 варианта решения: либо добавить 2 строки (есть закомментированные в исходниках пираний, не знаю озвучены ли они на русском), либо создать отдельную инстанцию таблички в башне Ксардаса и изменить условие диалога (плюс все остальные проверки в скриптах на каменные таблички)

- [ ] функция диалога `DIA_Xardas_WEAPON_Condition()`:
в 28-й версии было `if(!Npc_KnowsInfo(other,DIA_Xardas_TODO) && !Npc_KnowsInfo(other,DIA_Xardas_FirstEXIT) && (Kapitel < 3))`
в 29-й версии стало `if (Kapitel < 3)` как в исходниках игры
очевидно, что в определённый момент кто-то попытался поправить логику условия, но скорее всего допустил ошибку:
вместо `!Npc_KnowsInfo(other,DIA_Xardas_TODO)` должно было быть `Npc_KnowsInfo(other,DIA_Xardas_TODO)`,
Разговор будет звучать более естественно, если ГГ спросит об оружии **после** того, как узнает, что нужно сделать, **но до того**, как получит возможность осмотреться в башне Ксардаса.
Поэтому я бы поставил условие `if(Npc_KnowsInfo(other,DIA_Xardas_TODO) && !Npc_KnowsInfo(other,DIA_Xardas_FirstEXIT))`, **без** `(Kapitel < 3)`

- [ ] функция условия `DIA_Xardas_WhereEx_Condition()`:
`(Kapitel < 3)` лучше заменить на `(PlayerEnteredCity == FALSE)`, можно ещё добавить пару альтернативных условий, типа разговора с Малетом или Лобартом

- [ ] функция условия `DIA_Xardas_EQUIPMENT_Condition()`:
`if(Npc_KnowsInfo(other,DIA_Xardas_WEAPON) && Npc_KnowsInfo(other,DIA_Xardas_FirstEXIT) && (Kapitel < 3))` лучше заменить на
`if(Npc_KnowsInfo(other,DIA_Xardas_FirstEXIT) && (PlayerEnteredCity == FALSE))`

- [ ] функция условия `DIA_Xardas_Weiter_Condition()`:
`if(Npc_KnowsInfo(other,DIA_Xardas_FirstEXIT) && (Kapitel < 3))` заменить на
`if(Npc_KnowsInfo(other,DIA_Xardas_ARTEFAKT) && (Kapitel < 3))` - Ксардас рассказал зачем игроку Глаз Иноса,
тогда можно убрать условие `if(XardasKnowsAboutProof == FALSE)` в функции диалога `DIA_Xardas_Weiter_Info()`

**Comment by dosinabox** [comment]
Перемещено в обсуждения для удобства обсуждения.

**Comment by dosinabox** [comment]
> добавить 2 строки (есть закомментированные в исходниках пираний, не знаю озвучены ли они на русском)

Не озвучены ни в одной языковой версии:
```cpp
//AI_Output	(self, other, "DIA_Addon_Xardas_StonePlate_14_01"); //Ich habe so eine unten in der Höhlen vor meinem Turm gefunden.
//AI_Output	(self, other, "DIA_Addon_Xardas_StonePlate_14_02"); //Sie sind in einer seltsam alten Sprache geschrieben, die ich in dieser Form noch nirgendwo gesehen habe.
```

**Reply by SpectresHub** [reply]
В немецкой локализации озвучены

**Reply by dosinabox** [reply]
И правда. Надо будет завести такой список для разных языков, у меня есть только для русского и английского.
